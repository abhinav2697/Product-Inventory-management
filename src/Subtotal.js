import React from 'react';
import CurrencyFormat from 'react-currency-format';
import './Subtotal.css';


import { connect } from "react-redux";

export function Subtotal(props) {
    return (
        <div className="subtotal">
            <CurrencyFormat
                renderText={(value) => (
                    <>
                        <p>
                            Subtotal ({props.basket.length} item): <strong>{` ${value}`} </strong>
                        </p>
                        
                    </>
                )}
                decimalScale={2}
                value={props.basketAmount}
                displayType={"text"}
                thousandSeparator={true}
                prefix={"‎₹"}
            />
            <button>Proceed to checkout</button>
        </div>
    )
}

const getDataFromStore = (state) =>{
    return {
        basket: state.userReducer.basket,
        basketAmount: state.userReducer.basketAmount,
    }
}

export default connect(getDataFromStore,null)(Subtotal);
