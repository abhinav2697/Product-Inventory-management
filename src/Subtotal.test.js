import React from 'react';
import { shallow ,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Subtotal} from './Subtotal';
import CurrencyFormat from 'react-currency-format';

configure({ adapter: new Adapter() });

describe("ComponentName", () => {
    it("should render my component", () => {
      const wrapper = shallow(<Subtotal />);
      expect(wrapper.getElements()).toMatchSnapshot();
    });
  });