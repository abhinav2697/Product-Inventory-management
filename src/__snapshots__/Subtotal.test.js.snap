// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentName should render my component 1`] = `
Array [
  <div
    className="subtotal"
  >
    <CurrencyFormat
      allowNegative={true}
      decimalScale={2}
      decimalSeparator="."
      displayType="text"
      fixedDecimalScale={false}
      isAllowed={[Function]}
      isNumericString={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      onValueChange={[Function]}
      prefix="‎₹"
      renderText={[Function]}
      suffix=""
      thousandSeparator={true}
      thousandSpacing="3"
      type="text"
    />
    <button>
      Proceed to checkout
    </button>
  </div>,
]
`;
